tools/check_examples/check.py

```diff
@@ -128,10 +150,19 @@ def list_example_directories() -> list[Path]:
     if not list_directories:
         iprint_fail("Unable to list example directories.", 0)
         sys.exit(1)
-    list_directories = list(filter(ignore, list_directories))
+    list_directories = list(filter(filter_ignore, list_directories))
     return sorted(list_directories)
 
 
+def list_changed_files() -> list[Path]:
+    """Return a list of changed or untracked files."""
+    result = subprocess.run(["git", "status", "-su"], capture_output=True, text=True, check=False)
```

Comment:
It would be more reliable to use GitPython instead of parsing the output from a command line utility - console output can change without warning.  Something to consider doing another time.

Reply:
Will consider doing another time as per our discussion.